### PR TITLE
Run integration tests against local broker

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Uses the .NET Core AMQP broker for local E2E tests.
Removed tests that were specific to EH and SB.
Removed github workflow.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -54,9 +55,29 @@ jobs:
...
 
+      - task: UseDotNet@2
+        displayName: 'Use .NET sdk'
+        inputs:
+          packageType: sdk
+          version: 5.0.x
+          installationPath: $(Agent.ToolsDirectory)/dotnet
+
+      - script: |
+          git clone https://github.com/jhendrixMSFT/azure-amqp $(Pipeline.Workspace)/azure-amqp
```
Comment (jhendrixMSFT) [inline azure-pipelines.yml:66]: Using my fork for now until I get the `/headless` switch merged upstream.

File: integration_test.go [hunk 2]
```diff
@@ -1,64 +1,52 @@
...
-	"github.com/Azure/azure-sdk-for-go/services/eventhub/mgmt/2017-04-01/eventhub"
-	"github.com/Azure/azure-sdk-for-go/services/servicebus/mgmt/2017-04-01/servicebus"
 )
 
+var localBrokerAddr string
+
 func init() {
 	// rand used to generate queue names, non-determinism is fine for this use
 	rand.Seed(time.Now().UnixNano())
+	localBrokerAddr = os.Getenv("AMQP_BROKER_ADDR")
```
Comment (jhendrixMSFT) [inline integration_test.go:24]: These tests only run if this env var is set.

File: integration_test.go [hunk 3]
```diff
@@ -283,61 +263,42 @@ func TestIntegrationRoundTrip_Buffered(t *testing.T) {
...
 }
 
 func TestIntegrationReceiverModeSecond(t *testing.T) {
-	queueName, queuesClient, cleanup := newTestQueue(t, "receiver-mode-second")
-	defer cleanup()
-
+	if localBrokerAddr == "" {
+		t.Skip()
+	}
+	t.Skip("test fails due to race condition")
```
Comment (jhendrixMSFT) [inline integration_test.go:301]: Tracked in https://github.com/Azure/go-amqp/issues/55

**Review by richardpark-msft — APPROVED** [review]
Can't see any reason we wouldn't want to do this :)

**Pull request merged**